app/models/miq_server/configuration_management.rb

```diff
@@ -36,10 +37,4 @@ def config_activated(data)
 
     save
   end
-
-  def reset_server_caches
-    worker_manager.sync_config
-    sync_assigned_roles
```

Comment:
I'm worried that the order of this matters, because roles are stored in config (for better or worse).  Might not matter though if both are synced.

Reply:
Yeah I was worried about that also, I didn't see anything in `worker_manager.sync_config` that looked like it had to be _before_ syncing the assigned roles but I might be missing something: https://github.com/ManageIQ/manageiq/blob/master/app/models/miq_server/worker_management/monitor/settings.rb#L15-L29